qemu/tests/usb_block_dev_hotplug.py

```diff
@@ -0,0 +1,104 @@
+import logging
+
+from virttest.qemu_devices import qdevices
+from virttest import data_dir
+from virttest import storage
+from virttest import utils_misc
+
+
+def run(test, params, env):
+    """
+    Test hotplug of usb block devices.
+
+    1) Boot up guest with usb block device
+    2) verify the usb block device
+    2) Unplug the usb device and verify
+    4) Hotplug block device and verify
+
+    :param test:   QEMU test object.
+    :param params: Dictionary with the test parameters.
+    :param env:    Dictionary with test environment.
+    """
+    def _verify_usb_device_in_monitor(device_type):
+        logging.info("Check usb device information in monitor")
+        output = str(vm.monitor.info("usb"))
+        if params["product"] not in output:
+            test.fail("Could not find %s" % device_type)
+
+    def _verify_usb_device_in_guest(session, device_serial):
+        logging.info("Check usb device information in guest")
+
+        # both Linux and Windows
+        def _check_usb_info():
+            output = session.cmd_output(params["chk_usb_info_cmd"],
+                                        float(params["cmd_timeout"]))
+            return (device_serial in output)
+        res = utils_misc.wait_for(_check_usb_info,
+                                  float(params["cmd_timeout"]),
+                                  text="Wait for getting usb device info")
+        if res is None:
+            test.fail("Could not find the usb device serial:[%s]" %
+                      device_serial)
+
+        # Linux only
+        if params.get("os_type") == "linux":
+            drive_path = utils_misc.get_linux_drive_path(session,
+                                                         device_serial)
+            if not drive_path:
+                test.error("Could not find [%s]" % drive_path)
+
+            def _check_disk():
+                output = session.cmd_output("fdisk -l",
+                                            float(params["cmd_timeout"]))
+                return (drive_path in output)
+            res = utils_misc.wait_for(_check_disk,
+                                      float(params["cmd_timeout"]),
+                                      text="Wait for getting disk info")
```

Comment:
Same question, `fdisk -l` should give the result for the first time you send the command, looks don't have to use `utils_misc.wait_for` to try this command many times.